src/Microsoft.Health.Fhir.Core.UnitTests/Features/Operations/Export/ExportProcessingJobTests.cs

```diff
@@ -133,16 +133,79 @@ public async Task GivenAnExportJob_WhenItFinishesAPageOfResultsAndAFollowupJobEx
             Assert.Equal(followUpJob, queueClient.JobInfos[0]);
         }
 
-        private string GenerateJobRecord(OperationStatus status, string failureReason = null)
+        [Theory]
+        [InlineData("Patient", "Observation", null, null)]
+        [InlineData(null, null, "range1", "range2")]
+        public async Task GivenAnExportJob_WhenItFinishesAPageOfResultsAndNewerParallelJobExists_ThenANewProgressJobIsQueued(string resourceType1, string resourceType2, string feedRange1, string feedRange2)
+        {
+            string progressResult = string.Empty;
+
+            Progress<string> progress = new Progress<string>((result) =>
+            {
+                progressResult = result;
+            });
+
+            var queueClient = new TestQueueClient();
+            var processingJob = new ExportProcessingJob(MakeMockJobWithProgressUpdate, queueClient);
+
+            var runningJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running, resourceType: resourceType1, feedRange: feedRange1));
+
+            var followUpJob = GenerateJobInfo(GenerateJobRecord(OperationStatus.Running, resourceType: resourceType2, feedRange: feedRange2));
+            followUpJob.Id = runningJob.Id + 1;
+            queueClient.JobInfos.Add(followUpJob);
+
+            var taskResult = await processingJob.ExecuteAsync(runningJob, progress, CancellationToken.None);
+
+            Assert.True(queueClient.JobInfos.Count == 2);
+            Assert.DoesNotContain(_progressToken, queueClient.JobInfos[0].Definition);
+            Assert.Equal(followUpJob, queueClient.JobInfos[0]);
```

Comment:
Can an assert be added to show the second job is a follow up to the running job? Also, the name of followUpJob maybe should be changed in this test since it is a follow up to a parallel job, not the running job.